presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcMetadataCache.java

```diff
@@ -48,7 +49,7 @@ public class JdbcMetadataCache
     public JdbcMetadataCache(JdbcClient jdbcClient, JdbcMetadataConfig config, JdbcMetadataCacheStats stats)
     {
         this(
-                newCachedThreadPool(daemonThreadsNamed("jdbc-metadata-cache" + "-%s")),
+                newFixedThreadPool(config.getMetadataCacheThreadPoolSize(), daemonThreadsNamed("jdbc-metadata-cache" + "-%s")),
```

Comment:
The thread pool created with the `newCachedThreadPool` will spin up threads on demand for the asyncReloading to occur. Why is a fixed size thread pool better here ?

Reply:
I think he is trying to limit the numbers of threads that can used for the refresh. 

The following is better. its still a cache thread pool, but we limit it
`new BoundedExecutor(newCachedThreadPool(daemonThreadsNamed("jdbc-metadata-cache" + "-%s")), config.getMetadataCacheThreadPoolSize()),`

Reply:
TIL